bin/p2-launch/main.go

```diff
@@ -55,7 +55,8 @@ func main() {
 		log.Fatalf("%s", err)
 	}
 
-	pod := pods.NewPod(manifest.ID(), types.NodeName(*nodeName), pods.PodPath(*podRoot, manifest.ID()))
+	podFactory := pods.NewFactory(*podRoot, types.NodeName(*nodeName))
+	pod := podFactory.NewPod(manifest.ID())
```

Comment:
I could see this being one line to make it clear this is the only use of this factory. unless you think there will be more later. I wouldn't insist on this before shipping though.